File: DevSupportAndroid/src/main/java/com/sup/dev/android/views/screens/SInterstitialProgress.java
package com.sup.dev.android.views.screens;

import android.widget.ProgressBar;

import com.sup.dev.android.R;
import com.sup.dev.android.libs.screens.Screen;
import com.sup.dev.java.tools.ToolsThreads;

public class SInterstitialProgress extends Screen {

    private final ProgressBar vProgress;

    public SInterstitialProgress() {
        this(false);
    }

    public SInterstitialProgress(boolean startNow) {
        super(R.layout.screen_interstitial_progress);

        vProgress = findViewById(R.id.progress);
        backStackAllowed = false;

        if (!startNow)
            vProgress.setVisibility(INVISIBLE);
        ToolsThreads.main(1000, () -> vProgress.setVisibility(VISIBLE));
    }
}

File: DevSupportAndroid/src/main/java/com/sup/dev/android/views/screens/SInterstitialProgress.kt
package com.sup.dev.android.views.screens

import android.view.View
import android.widget.ProgressBar
import com.sup.dev.android.libs.screens.Screen
import com.sup.dev.java.tools.ToolsThreads

class SInterstitialProgress constructor(startNow: Boolean = false) : Screen(R.layout.screen_interstitial_progress) {

    private val vProgress: ProgressBar

    init {

        vProgress = findViewById(R.id.progress)
        isBackStackAllowed = false

        if (!startNow)
            vProgress.visibility = View.INVISIBLE
        ToolsThreads.main(1000) { vProgress.visibility = View.VISIBLE }
    }
}
